src/vehicle.cpp

```diff
@@ -899,413 +867,206 @@ void vehicle::smash_security_system(){
+    if( has_part( "ENGINE" ) ) {
+        if( g->u.controlling_vehicle || ( remote && engine_on ) ) {
+            options.emplace_back( _( "Stop driving" ), keybind( "TOGGLE_ENGINE" ) );
+            actions.push_back( [&] { 
+                if( engine_on && has_engine_type_not( fuel_type_muscle, true ) ){
+                    add_msg( _( "You turn the engine off and let go of the controls." ) );
+                } else {
+                    add_msg( _( "You let go of the controls." ) );
+                }
+                engine_on = false;
+                g->u.controlling_vehicle = false;
+                g->setremoteveh( nullptr );
+            } );
 
-    // Toggle engine on/off, stop driving if we are driving.
-    if( has_engine ) {
-        if( g->u.controlling_vehicle || ( remotely_controlled && engine_on ) ) {
-            menu.addentry( toggle_engine, true, 'e', _("Stop driving") );
         } else if( has_engine_type_not(fuel_type_muscle, true ) ) {
-            menu.addentry( toggle_engine, true, 'e', engine_on ?
-                           _("Turn off the engine") : _("Turn on the engine") );
-        }
-        if( has_electronic_controls  ) {
-            menu.addentry( toggle_cruise_control, true, 'c', cruise_on ?
-                           _("Disable cruise control") : _("Enable cruise control") );
+            options.emplace_back( engine_on ? _( "Turn off the engine" ) : _( "Turn on the engine" ), keybind( "TOGGLE_ENGINE" ) );
+            actions.push_back( [&] { 
+                if( engine_on ) {
+                    engine_on = false;
+                    add_msg( _( "You turn the engine off." ) );
+                } else {
+                    start_engines();
+                }
+            } );
         }
     }
 
-    if( is_alarm_on && velocity == 0 && !remotely_controlled ) {
-        menu.addentry( try_disarm_alarm, true, 'z', _("Try to disarm alarm.") );
+    if( has_part( "HORN") ) {
+        options.emplace_back( _( "Honk horn" ), keybind( "SOUND_HORN" ) );
+        actions.push_back( [&]{ honk_horn(); } );
     }
 
-    if ( has_electronic_controls && !lights().empty() ) {
-        menu.addentry( toggle_lights, true, 'v', _( "Control vehicle lights" ) );
-        auto opts = lights();
-        if( std::any_of( opts.begin(), opts.end(),[]( const vehicle_part *e ) {
-            return e->info().has_flag( VPFLAG_CONE_LIGHT ) && !e->enabled;
-        } ) ) {
-            menu.addentry( enable_headlights, true, 'h', _( "Turn on headlights" ) );
-        }
-        if( std::any_of( opts.begin(), opts.end(),[]( const vehicle_part *e ) {
-            return e->enabled;
-        } ) ) {
-            menu.addentry( disable_lights, true, 'D', _( "Turn off all lights" ) );
+    auto add_toggle = [&]( const std::string &name, char key, const std::string &flag ) {
+        if( has_part( flag ) ) {
+            auto msg = string_format( has_part( flag, true ) ?_( "Turn off %s" ) : _( "Turn on %s" ), name.c_str() );
```

Comment:
You explicitly say "turn off/on" the part here. Meaning it would be very misleading if a mixed state ever happened.

You can easily extract a "is this part kind on" here and use it to homogenize the next state.